test/index.spec.ts

```diff
@@ -12,4 +12,26 @@ describe('resin-semver', () => {
 			});
 		});
 	});
+
+	describe('.major()', () => {
+		it('should correctly match valid semver versions', () => {
+			expect(semver.major('4.5.1')).to.equal(4);
+		});
+
+		it('should correctly match resinOS prefixed versions', () => {
+			expect(semver.major('Resin OS v2.0.5')).to.equal(2);
+			expect(semver.major('Resin OS 2.0.2+rev2')).to.equal(2);
+			expect(semver.major('Resin OS 2.0.0.rev1 (prod)')).to.equal(2);
+			expect(semver.major('Resin OS 2.0.0-rc5.rev1')).to.equal(2);
+		});
+
+		it('should return null when version is `null`', () => {
+			expect(semver.major(null)).to.equal(null);
+		});
+
+		it('should return null when the version is contains no valid semver value', () => {
+			expect(semver.major('My dev version')).to.equal(null);
+			expect(semver.major('Linux 14.04')).to.equal(null);
```

Comment:
Can we add a case for non-ResinOS prefixed string that contains a semver 3-part version somewhere, just to make it clear (and guarantee for the future), that we generally don't parse versions like that?

As a fun example (and a great explanation of why comparing other people's versions is scary) a little research earlier came up with OpenSuse Leap's versioning scheme. They're currently on version `42.3.20170726.72bbcf8`, [to be followed next year](https://www.phoronix.com/scan.php?page=news_item&px=openSUSE-Leap-15-Versioning) by `15.x.x`, of course. :sob: